Shakeb

segalmax opened the pull request:
Shake parameters additions, changes, bindings and tests.

Commit:
shaking_par related changes, bindings and tests

Commit:
Rename of pyhton bindings tests module

Commit:
Add shaking parametrs tests.

yosefm commented on liboptv/src/parameters.c:
```diff
@@ -337,3 +337,57 @@ int compare_mm_np(mm_np *mm_np1, mm_np *mm_np2)
+	if (	   fscanf(par_file, "%d\n", &(ret->seq_first)) 				== 0
+			|| fscanf(par_file, "%d\n", &(ret->seq_last)) 			== 0
+			|| fscanf(par_file, "%d\n", &(ret->max_shaking_points)) 	== 0
+			|| fscanf(par_file, "%d\n", &(ret->max_shaking_frames)) 	== 0 )
```
If we're going through the trouble of all the 'or's then we don't need a goto.

segalmax replied:
My logic is: If one of 'fcanf's returns 0 then the parameter was not read and it is considered an error.
What do I miss here?

yosefm replied:
Of course. My problem is not twith the or, it's with the goto. The goto is there to avoid nesting, but with the short-circuit 'or' you don't nest so much, so you can do a simple if/else.

yosefm commented:
parameters.pyx actually contains a bunch of changes to the multimedia stuff.

segalmax commented:
These are some minor changes in constructor's signature. Should I commit
these in a separate PR?

On Sat, Jun 13, 2015 at 8:25 PM, Yosef Meller <notifications@github.com>
wrote:

> parameters.pyx actually contains a bunch of changes to the multimedia
> stuff.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yosefm/openptv/pull/11#issuecomment-111734815>.
>

yosefm commented:
Yes, drop them from this PR, you can propose a general janitoring PR which includes this, later.

yosefm commented on py_bind/test/test_parameters_bindings.py:
```diff
@@ -0,0 +1,65 @@
+        
+        # now we have two identical ShakingParams objects (sp1 and sp2) for comparison
+        self.assertTrue(sp1==sp2)
+        self.assertFalse(sp1!=sp2)
```
What's the difference between this line and the one before? Same for the following occurrence on lines 59-60.

segalmax replied:
The comparison is performed through function defined in ShakingParams:
```PYTHON
 def __richcmp__ (ShakingParams self, ShakingParams other, operator):  
```
The third parameter is integer value to indicate the operation performed: 2  for '=' , 3 for '!=' (you must refer to them separately). 
For more info: http://docs.cython.org/src/userguide/special_methods.html#rich-comparisons

So in order to check the comparisons I must pass '==' as well as '!=' .

yosefm replied:
I see. This is not trivial, better add a comment explaining it.

yosefm commented on py_bind/test/test_parameters_bindings.py:
```diff
@@ -0,0 +1,65 @@
+        
+        
+if __name__ == "__main__":
+    #import sys;sys.argv = ['', 'Test.testName']
```
Remove commented-out code.

yosefm commented on py_bind/test/test_parameters_bindings.py:
```diff
@@ -0,0 +1,65 @@
+        # checking C read_shaking_par function
+        # read shaking parameters from a file and
+        # verify the parameters were read correctly according to contents of specified file 
+        sp1.read_shaking_par("/home/student/Desktop/Max_OpenPtv/" + 
```
Hardcoded path makes the test fail for anyone but you. We said you'll make a special `tests/testing_fodder/` directory for the Python stuff.
(The test passes when manually changing the path to point to the correct file).

yosefm commented:
Other than the line comments above, looks solid.

Commit:
Add testing_fodder to py_bind/test folder - this folder is identical to testing_fodder in liboptv/tests
Add changes and corrections in response to Yosef's comments on pull request #11

Commit:
Restore the changes in parameters.pyx to MultimediaParams class.
These changes will be commited in a relevant branch.

Commit:
remove commented out code, remove irrelevant comment

yosefm commented:
Same comment here on the testing_fodder directory: take only what you need to survive.
https://www.youtube.com/watch?feature=player_detailpage&v=G1CYg0vIygE#t=30

yosefm commented:
Other than that, PR seems fine. Fix it and I'll do a PR to upstream.

segalmax committed:
remove unnecessary files in py_bind/testing_fodder

segalmax committed:
remove MultimediaParams changes

segalmax committed:
remove MultimediaParams changes, attempt 3

segalmax commented:
Were all issues resolved in this PR or changes are still needed?

alexlib commented:
I think this PR starts from some different point in history - I cannot test it anymore using 
`make verify`


`[ 36%] Built target optv
[ 42%] Linking C executable check_fb
clang: warning: argument unused during compilation: '-pthread'
ld: library not found for -lcheck
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [tests/check_fb] Error 1
make[2]: *** [tests/CMakeFiles/check_fb.dir/all] Error 2
make[1]: *** [tests/CMakeFiles/verify.dir/rule] Error 2
make: *** [verify] Error 2`
